### PR TITLE
changed the fragment_profile_stats to Constraintlayout

### PR DESCRIPTION
**Pull request by gconnect**
**What this PR does**
- Changed the Layout from LinearLayout to ConstraintLayout

**Open questions and concerns**
UI remains the same before and after

### CODE REVIEW

**Comment by ujjwalagrawal17** [comment]
@gconnect please remove the nested constraint layout and modify the layout IDs. Constraint layouts IDs should start with cl_

**Comment by gconnect** [comment]
@ujjwalagrawal17 I have modified the layout ids. I left the nested Constraints because they are needed to differentiate AsStudent and ByStudent categories used to set the visibility.

**Comment by ujjwalagrawal17** [comment]
Cool. I will review it soon.

**Comment by gconnect** [comment]
@ujjwalagrawal17 I have fixed the conflict

File: app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/profile/view/ProfileStatsFragment.kt [hunk 1]
```diff
@@ -58,32 +59,32 @@ class ProfileStatsFragment : Fragment() {
...
-            tvCountArticleViews.text = asStudentDetails.individualArticleViews
-            tvCountCommonsUploads.text = asStudentDetails.individualUploadCount
+            tv_title_impact_by_student.text = text
+            tv_count_articles_created.text = asStudentDetails.individualArticlesCreated
+            tv_count_articles_edited.text = asStudentDetails.individualArticleViews
+            tv_count_total_edits.text = asStudentDetails.individualArticleCount
+            tv_count_student_editors.text = asStudentDetails.individualUploadCount
+            tv_count_words_added.text = asStudentDetails.individualWordCount
+            tv_count_article_views.text = asStudentDetails.individualArticleViews
+            tv_count_commons_uploads.text = asStudentDetails.individualUploadCount
```
Comment (ujjwalagrawal17) [inline app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/profile/view/ProfileStatsFragment.kt]: Change Naming Convention from 'snake_case' to 'camelCase' in all elements.

File: app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/profile/view/ProfileStatsFragment.kt [hunk 2]
```diff
@@ -58,32 +59,32 @@ class ProfileStatsFragment : Fragment() {
...
         } ?: run {
-            llAsStudent.visibility = GONE
-            llNotEnrolled.visibility = VISIBLE
+            cl_as_student.visibility = GONE
+            tv_not_enrolled.visibility = VISIBLE
```
Comment (ujjwalagrawal17) [inline app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/profile/view/ProfileStatsFragment.kt]: Change Naming Convention from 'snake_case' to 'camelCase' in all elements.

File: app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/profile/view/ProfileStatsFragment.kt [hunk 3]
```diff
@@ -58,32 +59,32 @@ class ProfileStatsFragment : Fragment() {
...
-            tvInstructorCountArticlesCreated.text = byStudentDetails.newArticleCount
-            tvInstructorCountArticlesEdited.text = byStudentDetails.articleCount
-            tvInstructorCountCommonsUpload.text = byStudentDetails.uploadCount
+            tv_title_impact_by_student.text = text
+            tv_instructor_count_words_added.text = byStudentDetails.wordCount
+            tv_instructor_count_references_added.text = byStudentDetails.referencesCount
+            tv_instructor_count_article_views.text = byStudentDetails.viewSum
+            tv_instructor_count_articles_created.text = byStudentDetails.newArticleCount
+            tv_instructor_count_articles_edited.text = byStudentDetails.articleCount
+            tv_instructor_count_commons_upload.text = byStudentDetails.uploadCount
```
Comment (ujjwalagrawal17) [inline app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/profile/view/ProfileStatsFragment.kt]: Change Naming Convention from 'snake_case' to 'camelCase' in all elements.

**Comment by gconnect** [comment]
okay will do that

File: app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/profile/view/ProfileStatsFragment.kt (hunk 1 above)
Comment (gconnect) [reply app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/profile/view/ProfileStatsFragment.kt]: I have changed the naming convention from snake_case to camelCase

File: app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/profile/view/ProfileStatsFragment.kt (hunk 2 above)
Comment (gconnect) [reply app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/profile/view/ProfileStatsFragment.kt]: I have changed the naming convention from snake_case to camelCase

File: app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/profile/view/ProfileStatsFragment.kt (hunk 3 above)
Comment (gconnect) [reply app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/profile/view/ProfileStatsFragment.kt]: I have changed the naming convention from snake_case to camelCase

File: app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/profile/view/ProfileStatsFragment.kt [hunk 4]
```diff
@@ -58,7 +59,7 @@ class ProfileStatsFragment : Fragment() {
         profileResponse?.asStudent?.let {
             val asStudentDetails: AsStudentDetails = it
             val text = "Total impact made by $username's students"
```
Comment (victorvicari) [inline app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/profile/view/ProfileStatsFragment.kt]: Can this hardcode text be moved to the `strings.xml` ? Once every text is removed, the Suppress annotation above can be removed.

Comment (gconnect) [reply app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/profile/view/ProfileStatsFragment.kt]: Thanks for your observation. I will work on it.

**Pull request closed without merging**